streaming/streaming.go

```diff
@@ -0,0 +1,258 @@
+	}
+	err = json.Unmarshal(cfgFile, &streamingCfg)
+	if err != nil {
+		return streamingCfg, fmt.Errorf("unable to unmarshall json due to: %v", err)
+	}
+	err = VerifyAndUpdateCfg(&streamingCfg, dbName)
+	if err != nil {
+		return streamingCfg, fmt.Errorf("streaming config is incomplete: %v", err)
+	}
+	return streamingCfg, nil
+}
+
+// LaunchStream populates the parameters from the streaming config and triggers a stream on Cloud Datastream.
+func LaunchStream(ctx context.Context, dbName, projectID string, datastreamCfg DatastreamCfg) error {
+	fmt.Println("Launching stream...")
+	dsClient, err := datastream.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("datastream client can not be created: %v", err)
+	}
+	defer dsClient.Close()
+	fmt.Println("Created client...")
+
+	mydb := &datastreampb.MysqlDatabase{
+		DatabaseName: dbName,
+	}
+	mysqlSrcCfg := &datastreampb.MysqlSourceConfig{
+		Allowlist: &datastreampb.MysqlRdbms{MysqlDatabases: []*datastreampb.MysqlDatabase{mydb}},
+	}
+	gcsDstCfg := &datastreampb.GcsDestinationConfig{
+		Path:       datastreamCfg.DestinationConnectionConfig.Prefix,
+		FileFormat: &datastreampb.GcsDestinationConfig_AvroFileFormat{},
+	}
+	srcCfg := &datastreampb.SourceConfig{
+		SourceConnectionProfileName: fmt.Sprintf("projects/%s/locations/%s/connectionProfiles/%s", projectID, datastreamCfg.SourceConnectionConfig.Location, datastreamCfg.SourceConnectionConfig.Name),
+		SourceStreamConfig:          &datastreampb.SourceConfig_MysqlSourceConfig{MysqlSourceConfig: mysqlSrcCfg},
+	}
+	dstCfg := &datastreampb.DestinationConfig{
+		DestinationConnectionProfileName: fmt.Sprintf("projects/%s/locations/%s/connectionProfiles/%s", projectID, datastreamCfg.DestinationConnectionConfig.Location, datastreamCfg.DestinationConnectionConfig.Name),
+		DestinationStreamConfig:          &datastreampb.DestinationConfig_GcsDestinationConfig{GcsDestinationConfig: gcsDstCfg},
+	}
+	streamInfo := &datastreampb.Stream{
+		DisplayName:       datastreamCfg.StreamDisplayName,
+		SourceConfig:      srcCfg,
+		DestinationConfig: dstCfg,
+		State:             datastreampb.Stream_RUNNING,
+		BackfillStrategy:  &datastreampb.Stream_BackfillNone{BackfillNone: &datastreampb.Stream_BackfillNoneStrategy{}},
+	}
+	createStreamRequest := &datastreampb.CreateStreamRequest{
+		Parent:   fmt.Sprintf("projects/%s/locations/%s", projectID, datastreamCfg.StreamLocation),
+		StreamId: datastreamCfg.StreamId,
+		Stream:   streamInfo,
+	}
+
+	fmt.Println("Created stream request..")
+
+	dsOp, err := dsClient.CreateStream(ctx, createStreamRequest)
+	if err != nil {
+		fmt.Printf("createStreamRequest: %+v\n", createStreamRequest)
+		return fmt.Errorf("cannot create stream: %v ", err)
+	}
+
+	_, err = dsOp.Wait(ctx)
+	if err != nil {
+		fmt.Printf("createStreamRequest: %+v\n", createStreamRequest)
+		return fmt.Errorf("datastream create operation failed: %v", err)
+	}
+	fmt.Println("Successfully created stream")
```

Comment:
Add details of stream created

Reply:
Added streamid and source dbname in the statement